Patch USB1's ENDPTSTAT access, ATDTW bit offset; add BAUD[RIDMAE]; fix PIT LDVAL mask

mciantyre opened the pull request:
The PR adds a few SVD patches discovered through prototyping, and past discussions. All changes are backwards compatible.

- Change USB's `ENDPTSTAT` to read-write. The reference manual indicates that we need to write to this register during a bus reset. See the "Bus Reset" section under the USB Operational Model >> Device Operational Model:

  > Clear all setup token semaphores by reading the Endpoint Status (USB_ENDPTSTAT) register and writing the same value back to the Endpoint Status (USB_ENDPTSTAT) register.

  I'll note that the memory map in the reference manuals also say `ENDPTSTAT` is read-only, and that the field definition says the bit is cleared by flushing. So, I'm not sure how that jives with the quote above.

- Add the `RIDMAE` field to the `BAUD` register in i.MX RT 1015 and 1021 LPUART peripherals. We noted the missing field in [split HAL discussions](https://github.com/imxrt-rs/imxrt-hal/issues/56#issuecomment-706817586). We showed that the field is valid for all BAUD registers by studying the reference manuals. I'll grab this patch now, since I was making the `ENDPTSTAT` patch for some other work.

  There is an incantation; thank you for the help! ~~I can't figure out the `svdtools` incantation to add the `RIDMAE_0` and `RIDMAE_1` enumerated values, which are available in the other SVDs. I've noted that as a TODO in the patch file. Until we add that, the most portable code should use `0` or `1` constants, instead of the enumerated values, when setting the bit:~~

  ```rust
  use imxrt_ral as ral;
  ral::modify_register!(ral::lpuart, my_lpuart, BAUD, RIDMAE: 1 /* instead of RIDMAE_1 */);
  ```

- Correct `USBCMD[ATDTW]` bit offset for 1021, 1051, 1052, 1061, 1062, and 1064 chips. It should be at bit offset 14;. Confirmed by studying each questionable chip's reference manual.
- Correct the LDVAL bitwidth for PIT peripherals on 1015 and 1021 chips. SVD incorrectly notes that the field is 24 bits when it's 32 bits. This would prevent complete utilization of the timer.

mciantyre committed:
Mark USB ENDPTSTAT as read-write

i.MX RT reference manuals indicate that we need to read and write to
this register during a bus reset. This commit applies an SVD patch
to change the register access.

mciantyre committed:
Generate Rust code with RW ENDPTSTAT

See the previous commit for more information.

mciantyre committed:
Patch RIDMAE in 1015, 1021 SVDs

Not sure why this field is missing. The field is valid across all
BAUD registers, for all considered variants. We found this when
discussing a split HAL.

imxrtral.py still doesn't correctly fold the LPUART peripherals
together across all devices. It's either because

- it treats the missing enumerated values as semantic, incompatible
  differences
- it's not ordering the fields by bit offset, so it treats them as
  difference (observed by comparing the generated code)

Might need to investigate further if we go with this approach.

mciantyre committed:
Generate code with RIDMAE

See previous commit for details.

AlyoshaVasilieva commented:
I think this is what you're aiming for:

```yaml
# Add RIDMAE to LPUARTx BAUD registers
LPUART*:
  BAUD:
    _add:
      RIDMAE:
        description: Receiver Idle DMA Enable
        bitOffset: 20
        bitWidth: 1
        access: read-write
    RIDMAE:
      RIDMAE_0: [0, "DMA request disabled"]
      RIDMAE_1: [1, "DMA request enabled"]
```

It generates this code for the 1015 and 1021, which seems to match the rest: 

```rust
    /// Receiver Idle DMA Enable
    pub mod RIDMAE {
        /// Offset (20 bits)
        pub const offset: u32 = 20;
        /// Mask (1 bit: 1 << 20)
        pub const mask: u32 = 1 << offset;
        /// Read-only values (empty)
        pub mod R {}
        /// Write-only values (empty)
        pub mod W {}
        /// Read-write values
        pub mod RW {

            /// 0b0: DMA request disabled
            pub const RIDMAE_0: u32 = 0b0;

            /// 0b1: DMA request enabled
            pub const RIDMAE_1: u32 = 0b1;
        }
    }
```

mciantyre committed:
Generate RIDMAE enumerated values

Thanks @AlyoshaVasilieva for the tip! svdtools can include enumerated
values when adding fields.

mciantyre commented:
Thank you, that did the trick!

With these enumerated values, the LPUART peripheral merges cleanly across all chip variants (imxrt-rs/imxrt-hal#90).

mciantyre committed:
Correct USBCMD[ATDTW] bit offset

The bit is correct in the 1011 and 1015 SVDs. But, it's incorrect
in the 1021, 1051, 1052, 1061, 1062, and 1064 SVDs. I confirmed
that bit offset 14 is correct by studying the 1020, 1050, and 1060
reference manuals (skipped the 1064 RM, assumed it's consistent).

Note that this patch will affect the 1011 and 1015 SVDs, since it's
included by default. But, it will have no effect.

Tried to source an updated 1062 SVD file, thinking it could have been
fixed upstream. But, I didn't find an updated SVD from developer.arm.
com. Didn't consider any other chip SVDs.

teburd commented:
awesome!

mciantyre committed:
Fix bitwidth for PIT LDVAL in 1015, 1021

Specified as 24 bits in the SVDs, but it's actually 32 bits.
Confirmed in reference manual.

mciantyre committed:
Re-generate 1015, 1021 PIT module

See the previous commit for changes.

mciantyre committed:
Remove superfluous LPUART fields, register blocks

Should have been removed in 74a8ee711e, but our build process doesn't
clean up old code.
